pyquil/api/_qpu.py

```diff
@@ -134,20 +142,34 @@ def __init__(
         self._last_results: Dict[str, np.ndarray] = {}
         self._memory_results: Dict[str, Optional[np.ndarray]] = defaultdict(lambda: None)
         self._quantum_processor_id = quantum_processor_id
-        self._endpoint_id = endpoint_id
-
-        self._use_gateway = use_gateway
+        if execution_options is None:
+            execution_options_builder = ExecutionOptionsBuilder.default()
+            if endpoint_id is not None:
+                execution_options_builder.connection_strategy(ConnectionStrategy.endpoint_id(endpoint_id))
+            execution_options = execution_options_builder.build()
+        self.execution_options = execution_options
 
     @property
     def quantum_processor_id(self) -> str:
         """ID of quantum processor targeted."""
         return self._quantum_processor_id
 
-    def execute(self, executable: QuantumExecutable, memory_map: Optional[MemoryMap] = None) -> QPUExecuteResponse:
+    def execute(
+        self,
+        executable: QuantumExecutable,
+        memory_map: Optional[MemoryMap] = None,
+        execution_options: Optional[ExecutionOptions] = None,
+        **__: Any,
+    ) -> QPUExecuteResponse:
         """
         Enqueue a job for execution on the QPU. Returns a ``QPUExecuteResponse``, a
         job descriptor which should be passed directly to ``QPU.get_result`` to retrieve
         results.
+
+        :param:
+            execution_options: An optional `ExecutionOptions` enum that can be used
+              to configure how the job is submitted and retrieved from the QPU. If unset
+              `ExecutionOptions.default()` will be used.
```

Comment:
```suggestion
              to configure how the job is submitted and retrieved from the QPU. If unset,
              an appropriate default will be used.
```

nit, since there is the endpoint_id branch

Reply:
Ah, yep, good catch.